enhancements/sig-architecture/64-placementStrategy/README.md

```diff
@@ -0,0 +1,335 @@
+
+```go
+type PlacementSpec struct {
+	// ClusterSets represent the ManagedClusterSets from which the ManagedClusters are selected.
+	// +optional
+	ClusterSets []string `json:"clusterSets,omitempty"`
+
+	// Predicates represent a slice of predicates to select ManagedClusters. The predicates are ORed.
+	// +optional
+	Predicates []ClusterPredicate `json:"predicates,omitempty"`
+            ...
+ 
+	// PlacementStrategy define affinity clusters as subset of the selected clusters and divide the selected clusters to groups.
+	// +optional
+	PlacementStrategy PlacementStrategy `json:"placementStrategy,omitempty"`
+}
+
+
+type ClusterGroupType string
+
+// MandatoryClusterGroup indicate that all clusters in the group must have successfully applied workloads. 
+const MandatoryClusterGroup  ClusterGroupType
+// ProgressiveClusterGroup indicate that clusters in the group should apply the workloads. 
+const ProgressiveClusterGroup ClusterGroupType
+
+type PlacementStrategy struct {
+	// Affinity clusters label selector. Affinity clusters will be placed in cluster group index 0 and it will have MandatoryClusterGroup type.
+	// +optional
+	AffinityClusterGroup metav1.LabelSelector `json:"affinityClusterGroup,omitempty"`
+
+	// ClusterGroupPercentage is a percentage of the total selected clusters that will divide the clusters to groups.
+	// ex; for a total 100 clusters selected, ClusterGroupPercentage equal to 20% will divide the selected clusters to
+	//     5 groups each group should have 20 clusters.
+	// +kubebuilder:clustersGroupPercentage:Enum=None;5%;10%;15%;20%;25%;30%;50%;100%
+	// +kubebuilder:default:=None
+	// +optional
+	ClusterGroupPercentage string `json:"clustersGroupPercentage,omitempty"`
+
+	// NumberOfClustersPerGroup is the max number of clusters per group. If ClusterGroupPercentage is used the NumberOfClustersPerGroup will be ignored.
+	// +kubebuilder:default:=100
+	// +optional
+	NumberOfClustersPerGroup int32 `json:"numberOfClustersPerGroup,omitempty"` 
+
+	// Cluster Group Type indicates the group type MandatoryClusterGroup or ProgressiveClusterGroup.
+	// +kubebuilder:default:=ProgressiveClusterGroup
+	// +optional
+	ClusterGroupType ClusterGroupType `json:"clustersGroupType"`
```

Comment:
How would two consumers of the same placement choose different rollout mechanisms?

Reply:
If I understand  what u mean it depend on the use-case. Not sure if there a case should be progressive sometimes and mandatory other time.

Reply:
since this is more a hint from placement vs something placement actually uses to dictate its output? It might be something we explain in examples for controllers, but do not include in the api.